src/app/shared/components/modal/modal.component.html

```diff
@@ -1,3 +1,6 @@
 <div class="modal">
   the very best modal
+  <div class="modal-body">
+    <button (click)="closeModal('modal');">X</button>
```

Comment:
Why do you pass 'modal' instead of $event?

Reply:
Change it to $event